ddl/generated_column.go

```diff
@@ -122,13 +122,19 @@ func findPositionRelativeColumn(cols []*table.Column, pos *ast.ColumnPosition) (
 
 // findDependedColumnNames returns a set of string, which indicates
 // the names of the columns that are depended by colDef.
-func findDependedColumnNames(colDef *ast.ColumnDef) (generated bool, colsMap map[string]struct{}) {
+func findDependedColumnNames(schemaName string, tableName string, colDef *ast.ColumnDef) (generated bool, colsMap map[string]struct{}, err error) {
 	colsMap = make(map[string]struct{})
 	for _, option := range colDef.Options {
 		if option.Tp == ast.ColumnOptionGenerated {
 			generated = true
 			colNames := FindColumnNamesInExpr(option.Expr)
 			for _, depCol := range colNames {
+				if depCol.Schema.O != "" && schemaName != "" && depCol.Schema.O != schemaName {
```

Comment:
Is it possible that the original name have different Case and should be compared as `depCol.Schema.L` instead or with `strings.EqualFold()` instead?

Reply:
Yes, you're right. The PR is updated.